src/service/stellar.ts

```diff
@@ -0,0 +1,48 @@
+import { Keypair, Server } from 'stellar-sdk';
+import { ServerApi } from 'stellar-sdk/lib/server_api';
+
+const server = new Server(import.meta.env.VITE_URL_HORIZON);
+
+export function getPublicKeyFromPrivateKey(privateKey: string): string {
+  try {
+    const keyPair = Keypair.fromSecret(privateKey);
+    return keyPair.publicKey();
+  } catch (e) {
+    throw new Error('The private key was invalid');
+  }
+}
+
+export function shortenPublicKey(publicKey: string): string {
+  const first = publicKey.substring(0, 5);
+  const last = publicKey.slice(-5);
+  return `${first}...${last}`;
+}
```

Comment:
I think we can do something like this here:`
```typescript
export function getShorted(publicKey: string): string {
    return string.slice(0, 4).concat('...').concat(string.substr(-4));
}
```

Also, this doesn't belong here, this function should be declared in a helper.